examples/tensorgraph/mnist.py

```diff
@@ -6,82 +6,42 @@
 
 import deepchem as dc
 import tensorflow as tf
-from deepchem.models.tensorgraph.layers import Layer, Input, Reshape, Flatten
+from deepchem.models.tensorgraph.layers import Layer, Input, Reshape, Flatten, Feature, Conv2d, MaxPool, Label
 from deepchem.models.tensorgraph.layers import Dense, SoftMaxCrossEntropy, ReduceMean, SoftMax
 
 train = dc.data.NumpyDataset(mnist.train.images, mnist.train.labels)
 valid = dc.data.NumpyDataset(mnist.validation.images, mnist.validation.labels)
 
-tg = dc.models.TensorGraph(
-    tensorboard=True, model_dir='/tmp/mnist', batch_size=1000)
-feature = Input(shape=(None, 784))
-tg.add_feature(feature)
-
 # Images are square 28x28 (batch, height, width, channel)
-make_image = Reshape(shape=(-1, 28, 28, 1))
-tg.add_layer(make_image, parents=[feature])
-
-
-class Conv2d(Layer):
-
-  def __init__(self, num_outputs, kernel_size=5, **kwargs):
-    self.num_outputs = num_outputs
-    self.kernel_size = kernel_size
-    super().__init__(**kwargs)
-
-  def __call__(self, *parents):
-    parent_tensor = parents[0].out_tensor
-    out_tensor = tf.contrib.layers.conv2d(
-        parent_tensor,
-        num_outputs=self.num_outputs,
-        kernel_size=self.kernel_size,
-        padding="SAME",
-        activation_fn=tf.nn.relu,
-        normalizer_fn=tf.contrib.layers.batch_norm)
-    self.out_tensor = tf.nn.max_pool(
-        out_tensor, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
-    return self.out_tensor
-
-
-conv2d_1 = Conv2d(num_outputs=32)
-tg.add_layer(conv2d_1, parents=[make_image])
+feature = Feature(shape=(None, 784), name="Feature")
+make_image = Reshape(shape=(-1, 28, 28, 1), in_layers=[feature])
 
-conv2d_2 = Conv2d(num_outputs=64)
-tg.add_layer(conv2d_2, parents=[conv2d_1])
+conv2d_1 = Conv2d(num_outputs=32, in_layers=[make_image])
+maxpool_1 = MaxPool(in_layers=[conv2d_1])
 
-flatten = Flatten()
-tg.add_layer(flatten, parents=[conv2d_2])
+conv2d_2 = Conv2d(num_outputs=64, in_layers=[maxpool_1])
```

Comment:
Following Keras's example, let's not use the keyword args in these examples. I like the idea of making examples maximally simple for newbies. More advanced users can decide to use `kwargs` to avoid bugs in mature code.